src/components/common/Cropper/index.tsx

```diff
@@ -0,0 +1,199 @@
+import Modal from '@/components/common/Modal';
+import { useObjectUrl } from '@/lib/utils';
+import Image from 'next/image';
+import { PointerEvent, useCallback, useRef, useState } from 'react';
+import styles from './style.module.scss';
+
+/** Height of the preview square. */
+const HEIGHT = 200;
+
+/** Promisified version of `HTMLCanvasElement.toBlob` */
+function toBlob(canvas: HTMLCanvasElement, type?: string, quality?: number): Promise<Blob | null> {
+  return new Promise(resolve => {
+    canvas.toBlob(resolve, type, quality);
+  });
+}
+
+type DragState = {
+  pointerId: number;
+  offsetX: number;
+  offsetY: number;
+  initLeft: number;
+  initTop: number;
+};
+
+interface CropperProps {
+  file: Blob | null;
+  aspectRatio: number;
+  circle?: boolean;
+  maxFileHeight: number;
+  maxSize?: number;
+  // eslint-disable-next-line no-unused-vars
+  onCrop: (file: Blob) => void;
+  // eslint-disable-next-line no-unused-vars
+  onClose: (reason: 'invalid-image' | 'cannot-compress' | null) => void;
```

Comment:
are these rules still supposed to be disabled here?

Reply:
Yeah, for some reason ESLint thinks the parameters are unused even though they're just part of the type signature. You can see the rule being disabled throughout the codebase